### PR TITLE
OSSM-3732: Add istio postsubmit jobs

### PR DESCRIPTION
**Pull request by leoieggli**

### CODE REVIEW

**Comment by yxun** [comment]
In maistra/istio-operator maistra-2.2 branch there is no  make target 'update-2.1-charts'. 
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/38817/rehearse-38817-pull-ci-maistra-istio-operator-maistra-2.2-update-charts-in-2-3/1653843042727104512

So remove that target in the maistra-2.2 branch post submit jobs

File: ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-2.2.yaml [hunk 1]
```diff
@@ -32,6 +32,29 @@ tests:
...
+    export GOFLAGS="-mod=vendor"
+    export XDG_CACHE_HOME="/tmp/cache"
+    export GOCACHE="/tmp/cache"
+    export GOMODCACHE="/tmp/cache"
+    ./tools/automator.sh \
+      -o maistra \
+      -r istio-operator \
+      -b maistra-2.4 \
+      -f /creds-github/token \
+      '-c make clean update-charts-2.3' \
```
Comment (jwendell) [inline ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-2.2.yaml:50]: 2.2?

File: ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-2.2.yaml [hunk 2]
```diff
@@ -32,6 +32,29 @@ tests:
...
+      '-t Automator: Update charts from previous versions' \
+      -m update_charts_from_previous_versions_2.3
```
Comment (jwendell) [inline ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-2.2.yaml:52]: 2.2?

**Review by jwendell — CHANGES_REQUESTED** [review]
I think it will be easier to review if you split this PR in two.

From what I can see, this PR incorporates OSSM-3732 and OSSM-3733.

File: ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-2.2.yaml (hunk 1 above)
Comment (yxun) [reply ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-2.2.yaml:50]: @jwendell See my comment above

In maistra/istio-operator maistra-2.2 branch there is no make target 'update-2.1-charts'. and
make: *** No rule to make target 'update-2.2-charts'. Stop.
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/38817/rehearse-38817-pull-ci-maistra-istio-operator-maistra-2.2-update-charts-in-2-3/1653843042727104512
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/38817/rehearse-38817-pull-ci-maistra-istio-operator-maistra-2.2-update-charts-in-2-3/1653847872472354816

So remove that targets in the maistra-2.2 branch post submit jobs

Comment (yxun) [reply ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-2.2.yaml:50]: oh, nvm, I see we have  update-charts-2.2 . Let me update that

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-istio-operator-maistra-2.2-update-charts-in-2-4

**Review by jwendell — CHANGES_REQUESTED** [review]
Saying again:




I think it will be easier to review if you split this PR in two.

From what I can see, this PR incorporates OSSM-3732 and OSSM-3733.

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-istio-maistra-2.4-istio-update-operator-2-4

**Review by jwendell — APPROVED** [review]

**Comment by yxun** [comment]
@jwendell I added postsubmit true in the last commit and the ci bot removed the lgtm label. 
Could you approve this PR again ? thanks.

**Pull request merged**